web/handlers.go

```diff
@@ -210,6 +212,10 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		c.SetServerBusyError()
 	}
 
+	if c.Err == nil && h.IsLocal && *c.App.Config().ServiceSettings.EnableLocalMode && r.RemoteAddr == "@" {
```

Comment:
In the case where `h.IsLocal && r.RemoteAddress != "@"` we should probably bail out and return an error, since unless `h` is a reused handler from the existing APIs it won't have a reason to check any permissions

Reply:
I've created a specific error to handle these cases, PTAL to https://github.com/mattermost/mattermost-server/pull/14333/commits/2896bd552436c703ab417cea9704ff2a1ac0b186

Reply:
Modified the handling of the local origin with https://github.com/mattermost/mattermost-server/pull/14333/commits/309d207b9f78d07e1c040fd11e7577f0396604b8. Looks like `RemoteAddr` is only set to `@` in Linux, so this commit checks for the `IP:PORT` to decide if our connection is local instead